src/components/flash/flash.js

```diff
@@ -133,9 +135,16 @@ class Flash extends React.Component {
    * @method componentWillReceiveProps
    * @return(Void)
    */
-  componentWillReceiveProps(prevProps) {
-    if (prevProps.open !== this.props.open) {
-      this.setState({ dialogs: {} });
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open === this.props.open) { return; }
+
+    if (nextProps.open) {
+      this.setState({ dialogs: {}, open: nextProps.open });
```

Comment:
you don't need to set all state params on `setState`, just the ones you want to update